### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by mkevenaar** [mkevenaar/NickNelson#35]
TypeError: Cannot read properties of undefined (reading 'send')
    at Module.execute (file:///usr/src/bot/src/events/guildMemberAdd.js:64:25)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by davidzwa** [comment]
We need to know what the input variables on this line are:
```
let welcomeChannel = await client.tools.resolveChannel(channelName, guild);
```

Because the code itself looks ok. Maybe somewhere before `member.guild` or `guildData?.addons?.welcome.channel === channelName` is not set?

**Comment by davidzwa** [comment]
Lets converse in English 😉 

Yep validating this on startup (and otherwise crashing hard immediately) sounds best!

**Comment by mkevenaar** [comment]
Translated! :)

**Comment by mkevenaar** [comment]
:tada: This issue has been resolved in version 0.1.0 :tada:
